### PR TITLE
[#552] Compose dependencies update

### PR DESCRIPTION
**Pull request by HonzaR**
This PR updates several Compose-connected dependencies, and makes a few minor changes, see commits description.

Closes https://github.com/zcash/secant-android-wallet/issues/552, https://github.com/zcash/secant-android-wallet/issues/553.

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Manual tests: Did you update the [manual tests](../blob/main/docs/testing/manual_testing) as appropriate? _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed._ Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules.
- [ ] Documentation: Did you update documentation as appropriate? (e.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes?
- [ ] Screenshots: Did you provide before and after UI screenshots in the description of this pull request?  _This is only applicable for changes that modify the UI._
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?
- [ ] How is code coverage affected? _We encourage you to compare coverage before and after changes and when possible, leaving it in a better place._
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/scan/view/ScanView.kt [hunk 1]
```diff
@@ -194,22 +196,27 @@ private fun ScanMainContent(
...
-    if (!permissionState.hasPermission) {
+    if (!permissionState.status.isGranted) {
         setScanState(ScanState.Permission)
-        LaunchedEffect(key1 = UUID.randomUUID()) {
-            permissionState.launchPermissionRequest()
+        if (permissionState.status.shouldShowRationale) {
+            // keep blank screen with a link to the app settings
+            // user denied the permission previously
+        } else {
+            LaunchedEffect(key1 = true) {
```
Comment (ccjernigan) [inline ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/scan/view/ScanView.kt:213]: I have a concern this constant key might introduce a different bug.  Did you run through our manual test cases for this?

**Review by ccjernigan — COMMENTED** [review]
During our call today, you mentioned finding a bug where the permission prompt could appear after backing to the Home navigation target.  Is that a corner case worth adding to our manual test cases?

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/scan/view/ScanView.kt (hunk 1 above)
Comment (HonzaR) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/scan/view/ScanView.kt:214]: I've been testing several corner cases, see the first attached screen record below. Current implementation seems to be working well imho. Can you see any unwanted behaviour there? On the second attached screen record  I've tried to cover the cases where we probably want to show the "middle state" rational popup, as we've talked about yesterday. I also improved manual test for this screen, see last two commits. Thank you. 

https://user-images.githubusercontent.com/9017390/191736147-7f92188b-f8c7-4c93-aaf8-c5493c752624.mp4


https://user-images.githubusercontent.com/9017390/191736379-a325abd6-ecac-4e6a-b3cc-c67434e88ff9.mp4

Comment (ccjernigan) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/scan/view/ScanView.kt:214]: These videos really help—thanks for capturing them!

**Review by ccjernigan — APPROVED** [review]
Looks great; thanks for resolving the Compose issues.

I bumped the dependencies one more time since they got updated again yesterday.

We might have a failing robo test case; let's see if it still fails with the build that just got started.

**Comment by HonzaR** [comment]
> Looks great; thanks for resolving the Compose issues.
> 
> I bumped the dependencies one more time since they got updated again yesterday.
> 
> We might have a failing robo test case; let's see if it still fails with the build that just got started.

Yes, I can see in Testlab that some of the [matrixes](https://console.firebase.google.com/u/4/project/android-firebase-b0cf1/testlab/histories/bh.3fbf3cd2cc7177b7) failed. They failed for SDK 27 device only with the same exception:

` java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.List android.hardware.camera2.params.OutputConfiguration.getSurfaces()' on a null object reference`

I'm gonna file a new issue for that.

**Comment by ccjernigan** [comment]
> > Looks great; thanks for resolving the Compose issues.
> > I bumped the dependencies one more time since they got updated again yesterday.
> > We might have a failing robo test case; let's see if it still fails with the build that just got started.
> 
> Yes, I can see in Testlab that some of the [matrixes](https://console.firebase.google.com/u/4/project/android-firebase-b0cf1/testlab/histories/bh.3fbf3cd2cc7177b7) failed. They failed for SDK 27 device only with the same exception:
> 
> ` java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.List android.hardware.camera2.params.OutputConfiguration.getSurfaces()' on a null object reference`
> 
> I'm gonna file a new issue for that.

Sounds good.  We shouldn't let that block merging, as it seems like a test lab issue.
